include/envoy/event/dispatcher.h

```diff
@@ -202,6 +202,13 @@ class Dispatcher : public DispatcherBase {
                          Network::Address::InstanceConstSharedPtr source_address,
                          Network::TransportSocketPtr&& transport_socket,
                          const Network::ConnectionSocket::OptionsSharedPtr& options) PURE;
+  /**
+   * Register an internal listener manager for this dispatcher.
+   */
+  virtual void
+  registerInternalListenerManager(Network::InternalListenerManager& internal_listener_manager) PURE;
+
+  virtual Network::InternalListenerManagerOptRef getInternalListenerManager() PURE;
```

Comment:
Right now there are no non-test uses of these functions.  It is hard to understand how this mechanism will relate to the rest of the system.

If this object is meant to be a singleton or thread-local-singleton or factory, providing access to it via the dispatcher doesn't seem like the best way to do it.  Let's use thread local or singletons or factories instead.  I would suggest removing the changes to the dispatcher interface for now, and only add them in the PR that plugs everything together and more clearly shows why this needs to be part of the dispatcher interface.

Reply:
Thank you for the suggestion! thread-local-singleton might be the best if dispatcher doesn't accept this register-and-get semantic. 

Thread local single may have life time issue but it's good for now.

Factory and single is probably not good since the implemented listenermanger is connection handler, which is per-worker object now and in the future.

Is there an example of thread-local-singleton?

Reply:
My suggestion for now is to undo the dispatcher API changes and prototype alternate approaches to plug everything together in a followup PR that fully plugs in internal connections via the factory framework.

Reply:
I revert dispatcher API and deleted the new thin test about dispatcher
Merging conflict...